src/main/reactjs/src/redux/selectors/clerkTranslator.ts

```diff
@@ -87,26 +122,104 @@ const isAuthorisationExpiringSoon = (
 
 const filterByAuthorisationStatus = (
   translator: ClerkTranslator,
-  status: AuthorisationStatus
+  status: AuthorisationStatus,
+  currentDate: Date
 ) => {
-  const now = new Date();
-  switch (status) {
+  const expiringSoonDate = expiringSoonTreshold(currentDate);
+
+  return translator.authorisations.find((a) =>
+    matchesAuthorisationStatus(
+      { authorisationStatus: status },
+      currentDate,
+      expiringSoonDate,
+      a
+    )
+  );
+};
+
+const filterByAuthorisationCriteria = (
+  translator: ClerkTranslator,
+  filters: ClerkTranslatorFilter,
+  currentDate: Date
+) => {
+  const expiringSoonDate = expiringSoonTreshold(currentDate);
+
+  return translator.authorisations.find(
+    (a) =>
+      matchesFromLang(filters, a) &&
+      matchesToLang(filters, a) &&
+      matchesAuthorisationBasis(filters, a) &&
+      matchesAuthorisationStatus(filters, currentDate, expiringSoonDate, a)
+  );
+};
+
+const matchesFromLang = (
+  { fromLang }: ClerkTranslatorFilter,
+  authorisation: Authorisation
+) => {
+  if (fromLang) {
+    return authorisation.languagePair.from == fromLang;
+  }
+
+  return true;
+};
+
+const matchesToLang = (
```

Comment:
Tämän sekä vast. fromLang ja authorisationBasis perusteella filteröinnin voisi tehdä one-linereilla:
`return toLang ? authorisation.languagePair.to == toLang : true;`

Reply:
Sure. Will change.